### PR TITLE
feat: Add multiple bids execution

### PR DESCRIPTION
**Pull request by 0xShisui**

### CODE REVIEW

File: src/LooksRare.ts [hunk 1]
```diff
@@ -312,6 +312,55 @@ export class LooksRare {
     return execute(signer, this.addresses.EXCHANGE_V2, taker, maker, signature, merkleTree, referrer, overrides);
   }
 
+  /**
+   * Execute several orders
+   * @param orders List of orders data
+   * @param isAtomic Should the transaction revert or not if a trade fails
+   * @param referrer Referrer address
```
Comment (0xhiroshi) [inline src/LooksRare.ts:319]: affiliate?

File: src/__tests__/looksrare/executeMultipleTakerBids.test.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import { expect } from "chai";
+import { utils } from "ethers";
+import { ethers } from "hardhat";
+import { setUpContracts, SetupMocks, getSigners, Signers } from "../helpers/setup";
+import { LooksRare } from "../../LooksRare";
+import { SupportedChainId, CollectionType, StrategyType, CreateMakerInput } from "../../types";
+
+describe("execute taker bid", () => {
```
Comment (0xhiroshi) [inline src/__tests__/looksrare/executeMultipleTakerBids.test.ts:8]:
```suggestion
describe("execute multiple taker bids", () => {
```

File: src/__tests__/looksrare/executeMultipleTakerBids.test.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    const { maker: maker2 } = await lrUser1.createMakerAsk({ ...baseMakerAskInput, itemIds: [1], orderNonce: 1 });
+    const signature2 = await lrUser1.signMakerOrder(maker2);
+    const taker2 = lrUser2.createTaker(maker2, signers.user2.address);
+
+    const orders = [
+      { maker: maker1, taker: taker1, signature: signature1 },
+      { maker: maker2, taker: taker2, signature: signature2 },
+    ];
+
+    const contractMethods = lrUser2.executeMultipleOrders(orders, true);
```
Comment (0xhiroshi) [inline src/__tests__/looksrare/executeMultipleTakerBids.test.ts:54]: can you also test with false

File: src/__tests__/looksrare/executeMultipleTakerBids.test.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    const estimatedGas = await contractMethods.estimateGas();
+    expect(estimatedGas.toNumber()).to.be.greaterThan(0);
+
+    await expect(contractMethods.callStatic()).to.eventually.be.fulfilled;
+
+    tx = await contractMethods.call();
+    const receipt = await tx.wait();
+    expect(receipt.status).to.be.equal(1);
+  });
```
Comment (0xhiroshi) [inline src/__tests__/looksrare/executeMultipleTakerBids.test.ts:64]: leave a blank line between tests

File: src/__tests__/looksrare/executeMultipleTakerBids.test.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    const contractMethods = lrUser2.executeMultipleOrders(orders, true);
+
+    const estimatedGas = await contractMethods.estimateGas();
+    expect(estimatedGas.toNumber()).to.be.greaterThan(0);
+
+    await expect(contractMethods.callStatic()).to.eventually.be.fulfilled;
+
+    tx = await contractMethods.call();
+    const receipt = await tx.wait();
+    expect(receipt.status).to.be.equal(1);
```
Comment (0xhiroshi) [inline src/__tests__/looksrare/executeMultipleTakerBids.test.ts:63]: can you test the NFTs and ETH changed hands as well?

**Review by 0xhiroshi — APPROVED** [review]

File: src/LooksRare.ts [hunk 6]
```diff
@@ -296,20 +296,69 @@ export class LooksRare {
...
+
+  /**
+   * Execute several orders
+   * @param orders List of orders data
+   * @param isAtomic Should the transaction revert or not if a trade fails
+   * @param affiliate Affiliate address
+   * @param overrides Call overrides
+   * @returns ContractMethods
+   */
+  public executeMultipleOrders(
```
Comment (wjpg) [inline src/LooksRare.ts:323]: Just wondering why you didn't call this `executeMultipleTakerBids`?

Comment (0xShisui) [reply src/LooksRare.ts:323]: Because support for multiple taker asks is coming, and this function will follow the same pattern than `executeOrder`

**Pull request merged**